### DISCUSSION TITLE
dubbo3(http1\http2\tri\grpc)+higress

### DISCUSSION THREAD

**Discussion by zhangyungen** [apache/dubbo#13886] [General]
![Snipaste_2024-01-12_13-34-50](https://github.com/apache/dubbo/assets/38905541/61507b38-3113-499a-9c9f-6dbc12144b4a)

希望dubbo3 增加proxy gateway配置模块，促进dubbo成为微服务首选框架。如此便能快速支持跨语言调用，服务治理，服务熔断降级等功能。

**Comment by chickenlj** [comment]
Please check the features and best practices we are working on with the upcoming dubbo 3.3 release.

https://dubbo-next.staged.apache.org/zh-cn/overview/mannual/java-sdk/tasks/gateway/triple/

> Please notice that contents in the above link are all in experimental or developing phase.

**Comment by zhangyungen** [comment]
> Please check the features and best practices we are working on with the upcoming dubbo 3.3 release.
> 
> https://dubbo-next.staged.apache.org/zh-cn/overview/mannual/java-sdk/tasks/gateway/triple/
> 
> > Please notice that contents in the above link are all in experimental or developing phase.

明白谢谢，3.3 release 将实现triple 的网关代理，但是没有实现dubbo-provider之间调用的网关代理。

我的意思是，多个dubbo-provider之间的rpc流量也可通过微服务网关代理，这样所有服务间流量都经过微服务网关，微服务网关上就可以做很多服务治理与流量质量相关事项，类似sidecar模式，但是比sidecar轻量。如此这样dubbo-consumer 发起请求不是直接流向provider。而是变成  consumer   ----> gateway----->provider 。

如此，流量治理监控，服务治理，服务熔断、限流、降级，甚至业务监控，完全可以收拢在网关编排，或者增加扩展处理，在跨语言，跨框架的service+mesh中，减去了多套实现，多套管理。
